app/models/space.rb

```diff
@@ -40,28 +58,45 @@ def self.sort_by(sort_param)
     if sort_param == 'pri'
       sorted = sorted.reorder(:price_hour)
     elsif sort_param == 'rev'
-      sorted = sorted.select('COUNT(reviews.id) AS reviews_count').
-               joins('LEFT JOIN reviews ON spaces.id = reviews.id').
-               group(:id).reorder('reviews_count')
+      sorted = sorted.joins(:reviews).select('spaces.id, count(reviews.id) as number_of_reviews').group('spaces.id').order('number_of_reviews DESC')
```

Comment:
I agree when you say that "discards spaces with zero reviews may be confusing to the user if he sees some of the spaces disappearing after sorting them by number of reviews". A new solution using scope approach would be a better solution. I also agree that using LEFT JOIN instead of INNER JOIN solve this specific problem.